bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/ImageLoader.java

```diff
@@ -149,9 +152,63 @@ void reset() {
  * </ul>
  */
 public ImageData[] load(InputStream stream) {
+	return load(stream, 0);
+}
+
+/**
+ * Loads an array of <code>ImageData</code> objects from the
+ * specified input stream. If the stream is a SVG File and zoom is not 0,
+ * this method will try to rasterize the SVG.
+ * Throws an error if either an error occurs while loading the images, or if the images are not
+ * of a supported type. Returns the loaded image data array.
+ *
+ * @param stream the input stream to load the images from
+ * @param zoom the zoom factor to apply when rasterizing a SVG.
+ * A value of 0 means that the standard method for loading should be used.
+ * @return an array of <code>ImageData</code> objects loaded from the specified input stream
+ *
+ * @exception IllegalArgumentException <ul>
+ *    <li>ERROR_NULL_ARGUMENT - if the stream is null</li>
+ * </ul>
+ * @exception SWTException <ul>
+ *    <li>ERROR_IO - if an IO error occurs while reading from the stream</li>
+ *    <li>ERROR_INVALID_IMAGE - if the image stream contains invalid data</li>
+ *    <li>ERROR_UNSUPPORTED_FORMAT - if the image stream contains an unrecognized format</li>
+ * </ul>
+ *
+ * @since 3.129
+ */
+public ImageData[] load(InputStream stream, int zoom) {
 	if (stream == null) SWT.error(SWT.ERROR_NULL_ARGUMENT);
-	reset();
-	data = FileFormat.load(stream, this);
+    reset();
+    byte[] bytes = null;
+	try {
+		bytes = stream.readAllBytes();
+	} catch (IOException e) {
+		SWT.error(SWT.ERROR_IO, e);
+	}
+	ISVGRasterizer rasterizer = SVGRasterizerRegistry.getRasterizer();
+	if (rasterizer != null && zoom != 0) {
+	    try {
+	    	BufferedImage image = rasterizer.rasterizeSVG(bytes, zoom);
```

Comment:
Wouldn't it make sense to check whether the input is an SVG here instead of calling `rasterizeSVG` and rely on it to return `null` if is not an SVG? The current solution rather looks like a "try and error" approach. Since the `isSVG` functionality does not rely on any specific rendering implementation and library, it might make sense to provide that method in some `SVGUtil` that is provided directly by SWT, so that no concrete SVG rasterizer bundle is required to identify whether something is an SVG.

Reply:
I like this idea but in which fragment of SWT should this `SVGUtil` class be placed? It can't be used in GTK if I place it in win32.

Reply:
I will place it in Win32 for now as I want it in the official PR

Reply:
Great idea. I changed this.